### ISSUE TITLE
Large kmem_alloc(80000, 0x1000), please file an issue...

### ISSUE THREAD

**Issue by andrum993** [openzfs/zfs#15548]
### System information

Type | Version/Name
 --- | ---
Distribution Name	| Raspberry Pi OS (based on Debian)
Distribution Version	| Debian 12
Kernel Version	| 6.1.0-rpi6-rpi-2712
Architecture	| arm64
OpenZFS Version	| 2.1.11-1

### Describe the problem you're observing
dmesg filled with multiple traces telling me to file a bug here:

```
[ 1413.381286] Large kmem_alloc(80000, 0x1000), please file an issue at:
               https://github.com/openzfs/zfs/issues/new
[ 1413.381295] CPU: 0 PID: 783 Comm: arc_prune Tainted: P         C O       6.1.0-rpi6-rpi-2712 #1  Debian 1:6.1.58-1+rpt2
[ 1413.381299] Hardware name: Raspberry Pi 5 Model B Rev 1.0 (DT)
[ 1413.381301] Call trace:
[ 1413.381302]  dump_backtrace.part.0+0xec/0x100
[ 1413.381308]  show_stack+0x20/0x30
[ 1413.381310]  dump_stack_lvl+0x88/0xb4
[ 1413.381313]  dump_stack+0x18/0x34
[ 1413.381315]  spl_kmem_zalloc+0x10c/0x120 [spl]
[ 1413.381331]  zfs_prune+0x1a0/0x37c [zfs]
[ 1413.381481]  zpl_prune_sb+0x3c/0x70 [zfs]
[ 1413.381574]  arc_prune_task+0x2c/0x80 [zfs]
[ 1413.381666]  taskq_thread+0x2b8/0x4a0 [spl]
[ 1413.381679]  kthread+0xfc/0x110
[ 1413.381683]  ret_from_fork+0x10/0x20
[ 1413.393552] Large kmem_alloc(80000, 0x1000), please file an issue at:
               https://github.com/openzfs/zfs/issues/new
[ 1413.393559] CPU: 0 PID: 783 Comm: arc_prune Tainted: P         C O       6.1.0-rpi6-rpi-2712 #1  Debian 1:6.1.58-1+rpt2
[ 1413.393562] Hardware name: Raspberry Pi 5 Model B Rev 1.0 (DT)
[ 1413.393563] Call trace:
[ 1413.393565]  dump_backtrace.part.0+0xec/0x100
[ 1413.393571]  show_stack+0x20/0x30
[ 1413.393573]  dump_stack_lvl+0x88/0xb4
[ 1413.393577]  dump_stack+0x18/0x34
[ 1413.393579]  spl_kmem_zalloc+0x10c/0x120 [spl]
[ 1413.393604]  zfs_prune+0x1a0/0x37c [zfs]
[ 1413.393800]  zpl_prune_sb+0x3c/0x70 [zfs]
[ 1413.393892]  arc_prune_task+0x2c/0x80 [zfs]
[ 1413.393983]  taskq_thread+0x2b8/0x4a0 [spl]
[ 1413.393996]  kthread+0xfc/0x110
[ 1413.394000]  ret_from_fork+0x10/0x20
[ 1413.394826] Large kmem_alloc(80000, 0x1000), please file an issue at:
               https://github.com/openzfs/zfs/issues/new
[ 1413.394828] CPU: 0 PID: 783 Comm: arc_prune Tainted: P         C O       6.1.0-rpi6-rpi-2712 #1  Debian 1:6.1.58-1+rpt2
[ 1413.394830] Hardware name: Raspberry Pi 5 Model B Rev 1.0 (DT)
[ 1413.394831] Call trace:
[ 1413.394832]  dump_backtrace.part.0+0xec/0x100
[ 1413.394833]  show_stack+0x20/0x30
[ 1413.394835]  dump_stack_lvl+0x88/0xb4
[ 1413.394837]  dump_stack+0x18/0x34
[ 1413.394839]  spl_kmem_zalloc+0x10c/0x120 [spl]
[ 1413.394850]  zfs_prune+0x1a0/0x37c [zfs]
[ 1413.394948]  zpl_prune_sb+0x3c/0x70 [zfs]
[ 1413.395037]  arc_prune_task+0x2c/0x80 [zfs]
[ 1413.395127]  taskq_thread+0x2b8/0x4a0 [spl]
[ 1413.395139]  kthread+0xfc/0x110
[ 1413.395141]  ret_from_fork+0x10/0x20
etc.
```

### Describe how to reproduce the problem
Raspberry Pi 5 running Raspberry Pi OS Bookworm. Single zpool consisting of two SATA hard disks in USB 3 enclosures as mirrored pool. Running tar and zstd as part of custom backup script. Files on zpool being archived into tarball on another disk (USB3-connected SATA SSD):

```
 ZSTD_NBTHREADS='4' tar -c --zstd $BACKDIR/* -f $TARBALLDIR/pibackups.tar.zstd 2>/dev/null
```

I can see that there are similar issues reported already, however none have the specific allocation size reported on my system, hence this bug report.

**Comment by rincebrain** [comment]
#8491 fixed in 2.1.12 and newer.

**Comment by andrum993** [comment]
> #8491 fixed in 2.1.12 and newer.

Thanks - I've installed zfs-2.1.13-1~bpo12+1 from bookwork-backports which seems to have cured it 👍

**Comment by HankB** [comment]
Thanks for filing and fixing. I just ran into the exact same and am on my way to update ZFS from backports.

best,
